Woo Express: Added Plus plan to the Woo Express Plans page

nelsonec87 opened the pull request:
Added the "Plus" Plan to calypso so we can add it to the Plans page.

Added it as a fake plan, following the patterns from the `PLAN_ENTERPRISE_GRID_WPCOM` plan to have custom CtA, show the Logos, etc.

The CtA is redirecting users to `https://woocommerce.com/get-in-touch/`, as per p1680189727831979-slack-C0Q664T29

### Testing

* With a trial or Woo Express site, open the plans page (`/plans/site-slug`)
* Make sure the `plans/wooexpress-small` flag is enabled (`?flags=plans/wooexpress-small` in calypso live)
![image](https://user-images.githubusercontent.com/3801502/230595571-64ce109a-54f5-45e9-8579-5f6346da1df4.png)


Figma for reference:

![image](https://user-images.githubusercontent.com/3801502/229147798-0f17a6a0-3ff4-4ae6-bc80-9c9a30650123.png)

nelsonec87 committed:
Added plus plan to the trials Plans page

daledupreez commented on client/my-sites/plan-features-2023-grid/actions.tsx:
```diff
@@ -317,6 +319,25 @@ const PlanFeaturesActionsButton: React.FC< PlanFeaturesActionsButtonProps > = (
 		);
+	} else if ( isWooExpressPlusPlan ) {
+		const translateComponents = {
+			ExternalLink: (
```
Nit: given that we're only using this once, I would prefer this to be inline so it comes after the text. It looks like some of the other places where this is pushed into a `const` are because the values are used in two places.

daledupreez commented on client/my-sites/plan-features-2023-grid/index.tsx:
```diff
@@ -662,6 +666,7 @@ export class PlanFeatures2023Grid extends Component<
 						className={ getPlanClass( planName ) }
 						freePlan={ isFreePlan( planName ) }
 						isWpcomEnterpriseGridPlan={ isWpcomEnterpriseGridPlan( planName ) }
+						isWooExpressPlusPlan={ planName === PLAN_WOOEXPRESS_PLUS }
```
It might be worth creating an `isWooExpressPlusPlan()` helper function now.

daledupreez commented on packages/calypso-products/src/plans-list.tsx:
```diff
@@ -2262,6 +2279,16 @@ export const PLANS_LIST: Record< string, Plan | JetpackPlan | WPComPlan > = {
 	},
 
+	// Not a real plan. This is used to show the Plus offering in
+	// the main plans grid
```
Correction: this is not the main plans grid, it's the Woo Express plans grid. :grin:

daledupreez reviewed (commented):
Overall, I think this is a reasonable approach.

nelsonec87 committed:
Fixed button styles

nelsonec87 committed:
Added unit tests

nelsonec87 committed:
Added helper function

nelsonec87 committed:
Fixed comment

nelsonec87 committed:
Refactor button

nelsonec87 committed:
Added price copy to plus plan

nelsonec87 committed:
Use helper function

markbiek committed:
Misc rebase cleanup

markbiek commented:
The rebase on this was a little tricky but I think I got everything working properly. Another set of 👀 will be good for approval though.

sixhours commented:
@markbiek It looks like there are more conflicts to resolve; can we still test this or does it need to be rebased again?

markbiek commented:
> @markbiek It looks like there are more conflicts to resolve; can we still test this or does it need to be rebased again?

I'll rebase it again. Thanks for the heads-up!

markbiek commented:
@sixhours This should be ready for re-testing.

markbiek committed:
Fix lint error

aneeshd16 commented on client/my-sites/plan-features-2023-grid/index.tsx:
```diff
@@ -495,34 +496,42 @@ export class PlanFeatures2023Grid extends Component<
-					'has-border-top': ! isReskinned,
-				} );
-				const hasNoPrice = rawPrice === undefined || rawPrice === null;
+		const isLargeCurrency = planPropertiesObj.some(
+			( properties ) => properties?.rawPrice && properties?.rawPrice > 99000
+		);
```
These lines (501-503) were removed recently. I guess they got added here mistakenly when fixing a merge conflict. Can you please restore this to the version in trunk? 🙂

markbiek committed:
Fix stale reference to isLargeCurrency.

This was accidentally reintroduced during a rebase

aneeshd16 commented on client/my-sites/plan-features-2023-grid/index.tsx:
```diff
@@ -495,34 +496,38 @@ export class PlanFeatures2023Grid extends Component<
+					{ ! hasNoPrice && (
+						<PlanFeatures2023GridHeaderPrice
+							planProperties={ properties }
+							is2023OnboardingPricingGrid={ is2023OnboardingPricingGrid }
```
Thanks for fixing it! The value passed to the `PlanFeatures2023GridHeaderPrice` component should be `true`. The `is2023OnboardingPricingGrid` prop is never passed to this component despite being present in the type definition. It's confusing and we were also caught unawares by this. 🙂 

I have a fix for that issue here: https://github.com/Automattic/wp-calypso/pull/75826

markbiek replied:
Ah, thanks for catching that as well! I'll get that taken care of shortly.

sixhours commented:
I found a bug while I was testing this. Clicking on the button doesn't open the contact us page, but clicking on the text within the button does:

https://user-images.githubusercontent.com/2124984/232506416-b79d6e03-136a-4c19-8332-19c4fd29ddd5.mov

Otherwise everything looks as it should!

markbiek committed:
Fix stale reference to is2023OnboardingPricingGrid

This was accidentally reintroduced during a rebase

markbiek committed:
Fix Plus plan CTA button click area

The previous markup had an `a` tag inside of a `button` so there were
places where there was a gap surrounding the `a` tag and clicking there
didn't do anything.

Now we have just the `a` tag (which is more a11y-friendly anyway).

markbiek commented:
@sixhours That button issue should be resolved now.

aneeshd16 commented on client/my-sites/plan-features-2023-grid/index.tsx:
```diff
@@ -495,34 +496,38 @@ export class PlanFeatures2023Grid extends Component<
+		const { isReskinned, isLargeCurrency, translate } = this.props;
 
-		return planPropertiesObj
-			.filter( ( { isVisible } ) => isVisible )
```
Sorry, I did not notice this earlier! I was wondering if the removal of `.filter( ( { isVisible } ) => isVisible )` was intentional. Same for L750 below.

markbiek committed:
Fix missing filter for visible plan properties

aneeshd16 reviewed (approved):
Thanks for making the changes. This PR looks good to me from a pricing grid perspective. 👍

markbiek commented:
> Thanks for making the changes. This PR looks good to me from a pricing grid perspective. 👍

I super appreciate all the help reviewing. The rebase(s) on this ended up being super tricky.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7932178

Thank you @nelsonec87 for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7932178).
